WebDriverAgentLib/Categories/XCUIElement+FBTyping.m

```diff
@@ -71,27 +91,55 @@ - (BOOL)fb_typeText:(NSString *)text frequency:(NSUInteger)frequency error:(NSEr
 
 - (BOOL)fb_clearTextWithError:(NSError **)error
 {
-  if (0 == [self.value fb_visualLength]) {
+  id value = self.value;
+  if (nil != value && ![value isKindOfClass:NSString.class]) {
+    return [[[FBErrorBuilder builder]
+               withDescriptionFormat:@"The value of '%@' element is not a string and thus cannot be cleared", self.description]
+              buildError:error];
+  }
+  
+  if (nil == value || 0 == [value fb_visualLength]) {
+    // Short circuit if the content is not present
     return YES;
   }
 
   if (![self fb_prepareForTextInputWithError:error]) {
     return NO;
   }
   
-  NSUInteger preClearTextLength = 0;
-  NSData *encodedSequence = [@"\\u0008\\u007F" dataUsingEncoding:NSASCIIStringEncoding];
-  NSString *backspaceDeleteSequence = [[NSString alloc] initWithData:encodedSequence encoding:NSNonLossyASCIIStringEncoding];
-  while ([self.value fb_visualLength] != preClearTextLength) {
-    NSMutableString *textToType = @"".mutableCopy;
-    preClearTextLength = [self.value fb_visualLength];
-    for (NSUInteger i = 0 ; i < preClearTextLength ; i++) {
-      [textToType appendString:backspaceDeleteSequence];
+  static NSString *backspaceDeleteSequence;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    backspaceDeleteSequence = [[NSString alloc] initWithData:(NSData *)[@"\\u0008\\u007F" dataUsingEncoding:NSASCIIStringEncoding]
+                                                    encoding:NSNonLossyASCIIStringEncoding];
+  });
+  
+  NSUInteger retry = 0;
+  NSUInteger preClearTextLength = [value fb_visualLength];
+  do {
```

Comment:
Current logic returns error if the element has placeholder like the below case.
It happened after removing texts by back-key.

```
> e.clear
Selenium::WebDriver::Error::ElementNotInteractableError: Error Domain=com.facebook.WebDriverAgent Code=1 "Cannot clear the value of '"Search for a place or address" SearchField'" UserInfo={NSLocalizedDescription=Cannot clear the value of '"Search for a place or address" SearchField'}
from InvalidElementStateError: Error Domain=com.facebook.WebDriverAgent Code=1 "Cannot clear the value of '"Search for a place or address" SearchField'" UserInfo={NSLocalizedDescription=Cannot clear the value of '"Search for a place or address" SearchField'}
```

![image](https://user-images.githubusercontent.com/5511591/68909310-3e97a780-0791-11ea-8522-cd194d30ba7a.png)


How about changing the loop logic as an exact match as the `value` and current `self.value`?
In the case, here can go into loop logic when the backspace does not work. (But this case only can consider the exact match though) 

```diff
-  } while (preClearTextLength > 0);
+  } while ([self.value isEqualToString:value]);
```

Reply:
I've added the comparison with placeholder value. Hope, this helps